### ISSUE TITLE
 ConvolutionND output on GPU with certain batchsizes is zero

### ISSUE THREAD

**Issue by MannyKayy** [chainer/chainer#8100]
```py
> import chainer
> import chainer.links as L
```

CPU batchsize 64

```py
>> c =  L.ConvolutionND(3, 3, 64, ksize=3, stride=1, pad=1)
>> d = c.xp.random.randn(64,3,16,64,64).astype('f')
>> c(d).array.var()
0.9107105
>> c(d).array.mean()
-4.8422127e-05
```

Move to GPU

```py
>> c.to_gpu()

<chainer.links.connection.convolution_nd.ConvolutionND object at 0x7f99c0408990>

>> d = c.xp.random.randn(64,3,16,64,64).astype('f')
>> c(d).array.mean()
array(-0.00012456, dtype=float32)
>> c(d).array.var()
array(0.9110826, dtype=float32)
```

GPU batchsize = 32

```py
>> d = c.xp.random.randn(32,3,16,64,64).astype('f')
>> c(d).array.mean()
array(0., dtype=float32)
>> c(d).array.var()
array(0., dtype=float32)
```

GPU batchsize = 64

```py
>> d = c.xp.random.randn(64,3,16,64,64).astype('f')
>> c(d).array.mean()
array(-3.2548433e-06, dtype=float32)
>> c(d).array.var()
array(0.9115594, dtype=float32)
```

GPU batchsize = 32

```py
>> d = c.xp.random.randn(32,3,16,64,64).astype('f')
>> c(d).array.mean()
array(0., dtype=float32)
>> c(d).array.var()
array(0., dtype=float32)
>> c.xp.all(c(d).array == 0)
**array(True)**
```

Runtime info:

```
> chainer.print_runtime_info()
Platform: Linux-4.15.0-47-generic-x86_64-with-debian-buster-sid
Chainer: 7.0.0b3
ChainerX: Not Available
NumPy: 1.17.2
CuPy:
  CuPy Version          : 7.0.0b3
  CUDA Root             : /xxx/x/cuda/8.0
  CUDA Build Version    : 8000
  CUDA Driver Version   : 10010
  CUDA Runtime Version  : 8000
  cuDNN Build Version   : 7102
  cuDNN Version         : 7102
  NCCL Build Version    : 2213
  NCCL Runtime Version  : (unknown)
iDeep: Not Available
```

Default config info

```
> chainer.config.show()
_will_recompute                False
autotune                       False
cudnn_deterministic            False
cudnn_fast_batch_normalization False
debug                          False
dtype                          float32
enable_backprop                True
in_recomputing                 False
keep_graph_on_report           False
lazy_grad_sum                  False
schedule_func                  None
train                          True
type_check                     True
use_cudnn                      auto
use_cudnn_tensor_core          auto
use_ideep                      never
use_static_graph               True
warn_nondeterministic          False
```

**Comment by kmaehashi** [comment]
I couldn't reproduce this error (although I'm on CUDA 9.2 with cuDNN 7.5):

```py
import chainer
import chainer.links as L

for bs in (32, 64):
  for mode in ('cpu', 'gpu'):
    c = L.ConvolutionND(3, 3, 64, ksize=3, stride=1, pad=1)
    if mode == 'gpu':
      c.to_gpu()
    d = c.xp.random.randn(bs,3,16,64,64).astype('f')
    out = c(d)
    print('# bs = {}, array = {}'.format(bs, type(out.array)))
    print('var :', out.array.var())
    print('mean:', out.array.mean())
```

```
# bs = 32, array = <class 'numpy.ndarray'>
var : 0.9375737
mean: 4.1725e-05
# bs = 32, array = <class 'cupy.core.core.ndarray'>
var : 0.93067676
mean: -6.9142756e-05
# bs = 64, array = <class 'numpy.ndarray'>
var : 0.968008
mean: -0.00016163617
# bs = 64, array = <class 'cupy.core.core.ndarray'>
var : 0.9346749
mean: -7.527774e-05
```

Could you try reproducing it with cuDNN disabled?

**Comment by MannyKayy** [comment]
It appears to be a more nuanced issue as it doesn't correlate with batchsize.

First Run
```
>>> import chainer
>>> import chainer.links as L
>>>
>>> for bs in (32, 64, 32, 64, 64, 64):
...   for mode in ('cpu', 'gpu'):
...     c = L.ConvolutionND(3, 3, 64, ksize=3, stride=1, pad=1)
...     if mode == 'gpu':
...       c.to_gpu()
...     d = c.xp.random.randn(bs,3,16,64,64).astype('f')
...     out = c(d)
...     print('# bs = {}, array = {}'.format(bs, type(out.array)))
...     print('var :', out.array.var())
...     print('mean:', out.array.mean())
...
# bs = 32, array = <class 'numpy.ndarray'>
var : 0.95815516
mean: -3.072184e-05
<chainer.links.connection.convolution_nd.ConvolutionND object at 0x7f8560a06590>
# bs = 32, array = <class 'cupy.core.core.ndarray'>
var : 0.0
mean: 0.0
# bs = 64, array = <class 'numpy.ndarray'>
var : 0.9348984
mean: -1.8295983e-05
<chainer.links.connection.convolution_nd.ConvolutionND object at 0x7f8582dbfd50>
# bs = 64, array = <class 'cupy.core.core.ndarray'>
var : 0.94911224
mean: 1.723439e-05
# bs = 32, array = <class 'numpy.ndarray'>
var : 0.9432428
mean: 2.748484e-06
<chainer.links.connection.convolution_nd.ConvolutionND object at 0x7f85636fdc50>
# bs = 32, array = <class 'cupy.core.core.ndarray'>
var : 0.0
mean: 0.0
# bs = 64, array = <class 'numpy.ndarray'>
var : 0.9321733
mean: -1.6824411e-05
<chainer.links.connection.convolution_nd.ConvolutionND object at 0x7f8560869550>
# bs = 64, array = <class 'cupy.core.core.ndarray'>
var : 0.9569691
mean: 2.0605126e-05
# bs = 64, array = <class 'numpy.ndarray'>
var : 0.9169636
mean: 1.7625687e-05
<chainer.links.connection.convolution_nd.ConvolutionND object at 0x7f8564a21fd0>
# bs = 64, array = <class 'cupy.core.core.ndarray'>
var : 0.92691183
mean: -1.0163691e-05
# bs = 64, array = <class 'numpy.ndarray'>
var : 0.96403575
mean: -9.9605146e-05
<chainer.links.connection.convolution_nd.ConvolutionND object at 0x7f8560869110>
# bs = 64, array = <class 'cupy.core.core.ndarray'>
var : 0.9381752
mean: -8.15948e-05
```

Second run..
```
>>> import chainer
>>> import chainer.links as L
>>>
>>> for bs in (32, 64, 32, 64, 64, 64):
...   for mode in ('cpu', 'gpu'):
...     c = L.ConvolutionND(3, 3, 64, ksize=3, stride=1, pad=1)
...     if mode == 'gpu':
...       c.to_gpu()
...     d = c.xp.random.randn(bs,3,16,64,64).astype('f')
...     out = c(d)
...     print('# bs = {}, array = {}'.format(bs, type(out.array)))
...     print('var :', out.array.var())
...     print('mean:', out.array.mean())
...
# bs = 32, array = <class 'numpy.ndarray'>
var : 0.944605
mean: 0.00014973282
<chainer.links.connection.convolution_nd.ConvolutionND object at 0x7f8582dbfd50>
# bs = 32, array = <class 'cupy.core.core.ndarray'>
var : 0.0
mean: 0.0
# bs = 64, array = <class 'numpy.ndarray'>
var : 0.94555473
mean: 1.9031077e-05
<chainer.links.connection.convolution_nd.ConvolutionND object at 0x7f8560869550>
# bs = 64, array = <class 'cupy.core.core.ndarray'>
var : 0.9529564
mean: -4.3249725e-05
# bs = 32, array = <class 'numpy.ndarray'>
var : 0.9126329
mean: -0.00014837398
<chainer.links.connection.convolution_nd.ConvolutionND object at 0x7f859e27e790>
# bs = 32, array = <class 'cupy.core.core.ndarray'>
var : 0.0
mean: 0.0
# bs = 64, array = <class 'numpy.ndarray'>
var : 0.9625418
mean: -0.00025022958
<chainer.links.connection.convolution_nd.ConvolutionND object at 0x7f8560869b50>
# bs = 64, array = <class 'cupy.core.core.ndarray'>
var : 0.9465357
mean: 2.5492627e-05
# bs = 64, array = <class 'numpy.ndarray'>
var : 0.9364187
mean: 2.771529e-05
<chainer.links.connection.convolution_nd.ConvolutionND object at 0x7f859e27e750>
# bs = 64, array = <class 'cupy.core.core.ndarray'>
var : 0.9375669
mean: 6.180043e-05
# bs = 64, array = <class 'numpy.ndarray'>
var : 0.9354136
mean: -8.2954466e-05
<chainer.links.connection.convolution_nd.ConvolutionND object at 0x7f8560869a50>
# bs = 64, array = <class 'cupy.core.core.ndarray'>
var : 0.9447721
mean: 5.9716363e-05
'''

**Comment by MannyKayy** [comment]
@kmaehashi The issue appears to be remedied wit cuDNN disabled.

**Comment by kmaehashi** [comment]
In that case I guess it is a bug in cuDNN. Although I'm not tracking all of them there are many issues fixed since v7.1.2. Please consider upgrading it. https://docs.nvidia.com/deeplearning/sdk/cudnn-release-notes/rel_750.html#rel_750

**Comment by MannyKayy** [comment]
Thanks.. I will close the issue as this is non-chainer related bug.
